File: JavaScript/Queue/933. Number of Recent Calls.js
/* 933. Number of Recent Calls
Difficulty: Easy
Submission Link: https://leetcode.com/problems/number-of-recent-calls/submissions/1380151895/ */

var RecentCounter = function () {
    this.q = [];
};

RecentCounter.prototype.ping = function (t) {
    this.q.push(t);
    while (this.q[0] < t - 3000) {
        this.q.shift();
    }
    return this.q.length;
};

File: C#/Queue/933. Number of Recent Calls.cs
/* 933. Number of Recent Calls
Difficulty: Easy
Submission Link: https://leetcode.com/problems/number-of-recent-calls/submissions/1380152807/ */

public class RecentCounter
    {
        private Queue<int> _queue = new Queue<int>();
        public RecentCounter()
        {
            this._queue = new Queue<int>();
        }
        public int Ping(int t) 
        {
            _queue.Enqueue(t);
            while(_queue.Peek() < t - 3000)
            {
                _queue.Dequeue();
            }

            return _queue.Count;
        }
    }